wallet/src/de/schildbach/wallet/ui/send/SendCoinsBaseViewModel.kt

```diff
@@ -15,82 +15,47 @@
  */
 package de.schildbach.wallet.ui.send
 
-import android.os.Handler
-import android.os.HandlerThread
-import android.os.Looper
-import android.os.Process
-import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
-import de.schildbach.wallet.WalletApplication
 import de.schildbach.wallet.data.PaymentIntent
-import de.schildbach.wallet.livedata.Resource
-import de.schildbach.wallet.payments.DeriveKeyTask
-import de.schildbach.wallet.payments.SendCoinsOfflineTask
-import de.schildbach.wallet.security.SecurityGuard
-import org.bitcoinj.core.Coin
-import org.bitcoinj.core.Transaction
-import org.bitcoinj.crypto.KeyCrypterException
-import org.bitcoinj.utils.ExchangeRate
 import org.bitcoinj.utils.MonetaryFormat
 import org.bitcoinj.wallet.SendRequest
 import org.bitcoinj.wallet.Wallet
 import org.bitcoinj.wallet.ZeroConfCoinSelector
-import org.bouncycastle.crypto.params.KeyParameter
 import org.dash.wallet.common.Configuration
-import org.dash.wallet.common.Constants
-import org.dash.wallet.common.services.LeftoverBalanceException
-import org.dash.wallet.common.ui.dialogs.MinimumBalanceDialog
+import org.dash.wallet.common.util.Constants
+import org.dash.wallet.common.WalletDataProvider
 import javax.inject.Inject
 
 @HiltViewModel
 open class SendCoinsBaseViewModel @Inject constructor(
```

Comment:
The base viewModel is significantly lighter now since SendCoins and PaymentProtocol flows do not actually share that much of code.